core/variant_parser.cpp

```diff
@@ -1392,7 +1416,13 @@ static String rtosfix(double p_value) {
 	if (p_value == 0.0) {
 		return "0"; //avoid negative zero (-0) being written, which may annoy git, svn, etc. for changes when they don't exist.
 	} else {
-		return rtoss(p_value);
+		String ret = rtoss(p_value);
+		if (ret == "-inf") {
+			return "inf_neg";
+		} else if (ret == "-nan(ind)") {
```

Comment:
`rtoss` uses `sprintf`, and it can have more output variants depending on system / C runtime:

- C99 standard CRTs should output `inf`, `nan`, `nan(snan)`, `nan(ind)`.
- Old Visual Studio CRTs (MinGW may use it too) will output it as `1.#INF random-digits`, `.#IND random-digits` or `.#NAN random-digits`.

And any of these can be prefixed by `-`.

Reply:
Now I'm checking before stringifying the the value, but not sure could it be there any negative nan
@bruvzg

Reply:
I do not think there's any meaning to the sign of NaN if it is set, it's just artifact of previous operation. `isnan` should handle any NaN correctly and there's not reason to save it.

Reply:
What to do with NaN now?